### PR TITLE
renderer: disable VSYNC by default

### PR DESCRIPTION
**Pull request by espkk**
- TODO: make it configurable

### CODE REVIEW

File: src/apps/ENGINE/src/Main.cpp [hunk 1]
```diff
@@ -58,7 +58,7 @@ int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, PSTR szCmdLine,
         return EXIT_SUCCESS;
     }
 
-    SDL_InitSubSystem(SDL_INIT_EVENTS | SDL_INIT_VIDEO | SDL_INIT_GAMECONTROLLER);
```
Comment (np-c0mp1ete) [inline src/apps/ENGINE/src/Main.cpp:61]: Event subsystem is initialized by default, right?

**Review by np-c0mp1ete — COMMENTED** [review]
Why do we want to disable vsync?

File: src/apps/ENGINE/src/Main.cpp (hunk 1 above)
Comment (espkk) [reply src/apps/ENGINE/src/Main.cpp:61]: Right, it is implied by SDL_INIT_VIDEO

**Comment by espkk** [comment]
> 
> 
> Why do we want to disable vsync?

In general, it has a negative impact on fps. I know only a few people who actually use it.
But of course, this should be configurable as well as many other video settings. I plan to have a toml config in stash (documents) that may be configured manually until implemented in-game.
Also, HZ-based frame capping should be implemented (basically through setting specific maxFps automatically) for reasonable automatic frame capping without involving VSYNC

**Comment by np-c0mp1ete** [comment]
> > Why do we want to disable vsync?
> 
> In general, it has a negative impact on fps. I know only a few people who actually use it.
> But of course, this should be configurable as well as many other video settings. I plan to have a toml config in stash (documents) that may be configured manually until implemented in-game.
> Also, HZ-based frame capping should be implemented (basically through setting specific maxFps automatically) for reasonable automatic frame capping without involving VSYNC

I mean, what's the point in unlimited fps if the monitor refresh rate is limited, not even saying about the human eye? :)
IMHO it will just introduce frame tearing and make rendering look buggy.
But again, feel free to merge, if you really want it.

**Comment by espkk** [comment]
> 
> 
> > > Why do we want to disable vsync?
> > 
> > 
> > In general, it has a negative impact on fps. I know only a few people who actually use it.
> > But of course, this should be configurable as well as many other video settings. I plan to have a toml config in stash (documents) that may be configured manually until implemented in-game.
> > Also, HZ-based frame capping should be implemented (basically through setting specific maxFps automatically) for reasonable automatic frame capping without involving VSYNC
> 
> I mean, what's the point in unlimited fps if the monitor refresh rate is limited, not even saying about the human eye? :)
> IMHO it will just introduce frame tearing and make rendering look buggy.
> But again, feel free to merge, if you really want it.

VSYNC is just bad... it drastically increases latency and may screw up fps especially when the latter borderlines with HZ.
In my tests when I had fps drops from ~300to200 without VSYNC it resulted in ~144to110 drops with VSYNC enabled which is kind of distractive already.

As for me, I've never experienced screen tearing. But if it may appear we probably should try to involve triple buffering in windowed mode and gsync/adaptive sync or whatever it is called instead of vsync in fullscreen(do anyone indeed use fullscreen for non-fps games?).

Ok, maybe we should hold this until the optional setting is available, but I don't really want to introduce any new settings into engine.ini
So, consider this as blocked until I implement a new configuration system then

**Comment by nbohr1more** [comment]
In native Doom 3, if you set r_swapInterval -1 this passes the associated swap attribute to the GL driver to enable "Adaptive Sync" which offers better performance while still performing vsync when possible

**Comment by espkk** [comment]
> 
> 
> In native Doom 3, if you set r_swapInterval -1 this passes the associated swap attribute to the GL driver to enable "Adaptive Sync" which offers better performance while still performing vsync when possible

Yeah, that's what I meant 👍
But there are peculiarities depending on hardware and graphics API, so I cannot dive into it until higher prio tasks are done

**Pull request closed without merging**